Verse/src/DecoderDescriptors/Tree/IReader.cs

```diff
@@ -6,12 +6,12 @@ namespace Verse.DecoderDescriptors.Tree
 	internal interface IReader<TState, TNative, TKey>
 	{
 		BrowserMove<TElement> ReadToArray<TElement>(TState state, Func<TElement> constructor,
-			ReaderCallback<TState, TNative, TKey, TElement> callback);
+			ReaderCallback<TState, TNative, TKey, TElement> callback, out bool isVoid);
 
 		bool ReadToObject<TObject>(TState state, ILookupNode<TKey, ReaderCallback<TState, TNative, TKey, TObject>> root,

```

Comment:
It looks like return value and `isVoid` are correlated: `isVoid` means you won't take the entity into consideration, a bit like what the return code is saying. Also, returning false and defining `isVoid` is incompatible. Therefore I would suggest replacing the boolean output of these 3 methods by an enum with 3 different statuses:

- Failed (equivalent to returning "false")
- Succeeded (equivalent to returning "true" with "isVoid" = false)
- Ignored (equivalent to returning "true" with "isVoid" = true)

What do you think?

Reply:
I updated with it....